torch_geometric/utils/homophily.py

```diff
@@ -63,11 +76,27 @@ def homophily(edge_index: Adj, y: Tensor, batch: OptTensor = None,
         else:
             return scatter_mean(out, batch[col], dim=0)
 
-    else:
+    elif method == 'node':
         out = torch.zeros(row.size(0), device=row.device)
         out[y[row] == y[col]] = 1.
         out = scatter_mean(out, col, 0, dim_size=y.size(0))
         if batch is None:
             return float(out.mean())
         else:
             return scatter_mean(out, batch, dim=0)
+
+    else:
+        c = y.squeeze().max() + 1
+        nonzero_labels = y[y >= 0]
```

Comment:
Why do we need this exactly?

Reply:
It can be removed safely given that there are no negative labels